### PR TITLE
Hmusum/order zookeeper server list so localhost comes first

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

File: zkfacade/src/main/java/com/yahoo/vespa/curator/Curator.java [hunk 1]
```diff
@@ -69,17 +70,25 @@ public Curator(ConfigserverConfig configserverConfig, ZooKeeperServer server) {
...
-            if (i < config.zookeeperserver().size() - 1) {
-                sb.append(",");
-            }
+            servers.add(server.hostname() + ":" + server.port());
+            if (server.hostname().equals(hostName))
+                indexForThisConfigServer = i;
         }
-        return sb.toString();
+        // Rotate list of servers so that config server on this host is first in the list
+        Collections.rotate(servers, configServerCount - indexForThisConfigServer);
```
Comment (baldersheim) [inline zkfacade/src/main/java/com/yahoo/vespa/curator/Curator.java]: You should add an if(indexForThisConfigServer != 0)

File: zkfacade/src/test/java/com/yahoo/vespa/zookeeper/CuratorTest.java [hunk 2]
```diff
@@ -74,6 +75,18 @@ public void require_that_server_count_is_correct() {
         }
     }
 
+    @Test
+    public void require_that_server_order_is_correct() {
+        ConfigserverConfig.Builder builder = new ConfigserverConfig.Builder();
+        builder.zookeeperserver(createZKBuilder("localhost", port1));
+        builder.zookeeperserver(createZKBuilder(HostName.getLocalhost(), port2));
```
Comment (baldersheim) [inline zkfacade/src/test/java/com/yahoo/vespa/zookeeper/CuratorTest.java]: Use a longer list that 2 to ensure it is not testing that it is reversed.

**Review by baldersheim — CHANGES_REQUESTED** [review]

File: zkfacade/src/test/java/com/yahoo/vespa/zookeeper/CuratorTest.java (hunk 2 above)
Comment (hmusum) [reply zkfacade/src/test/java/com/yahoo/vespa/zookeeper/CuratorTest.java]: fixed

**Review by baldersheim — APPROVED** [review]

**Comment by baldersheim** [comment]
@hakonhall FYI

**Comment by baldersheim** [comment]
@bratseth FYI

**Comment by hakonhall** [comment]
the order in the connectstring doesn't matter. The list is shuffled by the ZooKeeper client.

**Comment by baldersheim** [comment]
Not in the code we looked at. At cfg3 was also always on top.

**Pull request merged**